File: src/main/java/funkies/errors/DataError.java
package funkies.errors;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

//200-299
@Getter
@RequiredArgsConstructor
public enum DataError implements SystemErrorCode {
    NOT_EXISTS(200, "Data with id [{}] doesn't exists"),
    NO_ACCESS(201, "Access to data is forbidden");

    final int code;

    final String message;

}

File: src/main/kotlin/kfunkies/errors/DataError.kt
package kfunkies.errors

enum class DataError(
    private val code: Int,
    private val message: String
) : SystemErrorCode {
    NOT_EXISTS(code = 200, message = "Data with id {} doesn't exists"),
    NO_ACCESS(code = 201, message = "Access to data is forbidden");

    override fun getCode() = code
    override fun id() = name
    override fun getMessage() = message
}